### DISCUSSION TITLE
Mapeamentos - EntityFramework

### DISCUSSION THREAD

**Discussion by DiegoOliver94** [balta-io/forum#168] [Tira Dúvidas] [answered]
Boa tarde.

Na aula "utilizando os mapeamentos criados" do curso de "
Fundamentos do Entity Framework" me gerou uma dúvida, por que role recebe um .HasOne<**Role**> e user um .HasOne<**User**>
![image](https://user-images.githubusercontent.com/98860594/207719918-2b7388b0-3c0f-4f2a-bc53-1acbf6755e68.png)

e na aula "Relacionamento muitos para muitos" uma tag recebe um .HasOne<**Posts**> e o post recebe um .HasOne<**Tags**>?
não consegui compreender se tem um padrão 

![image](https://user-images.githubusercontent.com/98860594/207721431-92334eed-e809-4afd-802b-7384977cd80e.png)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @DiegoOliver94, como vai?
Sempre que temos um relacionamento muitos para muitos, criamos uma terceira tabela, chamada tabela associativa e as chaves primárias vão para esta tabela, fazendo assim o relacionamento.
Então no fim, temos uma terceira tabela com dois relacionamentos um para um. Por isto o HasOne aí.

**Reply by DiegoOliver94** [reply]
entendi, neste caso este trecho post => post.HasOne<**Tag**>, deve ser post => post.HasOne<**Post**>, pois esta identificando que aquele objeto é do tipo Post e pertence a coluna PostId nesta terceira tabela criada, correto?

**Reply by andrebaltieri** [reply]
Exatamente!!! HasOne significa "têm um"... Então `post.HasOne<Tag>` interpretamos como "post tem uma tag".
HasMany significa "têm muitos", seria para o caso de listas.

**Reply by oGrandeK** [reply]
Professor acho que o senhor entendeu errado. Assim como eu também achei estranho que no PostMap você usa `post => post.HasOne<Tag>()` e no UserMap você usa `role => role.HasOne<role>()`.

A nossa dúvida está somente na sintaxe. Porquê eles estão diferente? Um não deveria ficar igual o outro? o UserMap não deveria ser `role.HasOne<user>` em vez de` role.HasOne<role>`?

Espero que eu tenha conseguido me explicar direito rs. :)
